python/snewpy/models/ccsn.py

```diff
@@ -54,32 +56,114 @@ class Nakazato_2013(PinchedModel):
     (2013), ApJ 804:75 (2015), PASJ 73:639 (2021). See also http://asphwww.ph.noda.tus.ac.jp/snn/.
     """
 
-    def __init__(self, filename):
-        """
+    param = {
+        'progenitor_mass': [13, 20, 30, 50] * u.Msun,
+        'revival_time': [0, 100, 200, 300] * u.ms,
+        'metallicity': [0.02, 0.004],
+        'eos': ['LS220', 'shen', 'togashi']
+    }
+
+    def __init__(self, filename=None, *, progenitor_mass=None, revival_time=None, metallicity=None, eos=None):
```

Comment:
How would this be presented to users? If `load()` is the main user-facing function, then the function signature in documentation would be quite puzzling; and referring them to `gen_filename()` partially defeats the purpose of hiding these implementation details from the user.

Also, if the function signature is simply `load(cls, **kwargs)`, this will make autocompletion more tricky since the parameter names would need to be extracted from a different function. (It’s possible that the overhead for this is not too bad; but it’d be good to check this.)

Reply:
I agree that it's worse than having `__init__(self, progenitor_mass, revival_time, metallicity, eos)` but still much better than `__init__(self, filename=None, *, progenitor_mass=None, revival_time=None, metallicity=None, eos=None)`.

> If `load()` is the main user-facing function, then the function signature in documentation would be quite puzzling; and referring them to `gen_filename()` partially defeats the purpose of hiding these implementation details from the user

We have the list of required parameters in the class' static variable `param`, so it can also be used for the documentation.
If that's not clear enough, we can, of course, define `load(cls, progenitor_mass, revival_time, metallicity, eos)` for each class. But that would be a lot of repeating code.


> this will make autocompletion more tricky

I don't see the difficulty for the autocompletion.